src/mlpack/methods/ann/loss_functions/triplet_margin_loss.hpp

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file triplet_margin_loss.hpp
+ * @author Prince Gupta
+ *
+ * Definition of the Triplet Margin Loss function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_ANN_LOSS_FUNCTION_TRIPLET_MARGIN_LOSS_HPP
+#define MLPACK_ANN_LOSS_FUNCTION_TRIPLET_MARGIN_LOSS_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class TripletMarginLoss
+{
+ public:
+  /**
+   * Create the TripletMarginLoss object with Hyperparameter margin.
+   */
+  TripletMarginLoss(const double margin = 1.0);
+
+  /**
+   * Computes the Triplet Margin Loss function.
+   *
+   * @param input The propagated input activation.
+   * @param target The target vector.
+   */
+template<typename AnchorType, typename PositiveType, typename NegativeType>
```

Comment:
This should align with * from the comments. This also true in other places as well.

Reply:
Let's keep it like this only, no need to reduce templates, I checked couple of other programs they use different template even though they must have same datatype.